Improvements for backup description and configuration for single node, cluster , quick start (#9459)
VictoriaMetrics/VictoriaMetrics: docs/victoriametrics/vmbackup.md

  - /vmbackup/index.html
  - /vmbackup/
---
## Quick start

> `vmbackup` creates VictoriaMetrics data backups to protect from user errors such as accidental data deletion.

Author: That was original wording from doc, but I like your changes

  - /vmbackup/index.html
  - /vmbackup/
---
## Quick start

`vmbackup` creates backups of VictoriaMetrics data to protect against hardware failures and accidental data loss.
Whether you are using a single-node  or a cluster version, it is recommended to use `vmbackup` to perform periodical data backup from instant snapshots.

> More information how to work with them could be found in [instant snapshots documentation](https://docs.victoriametrics.com/single-server-victoriametrics/#how-to-work-with-snapshots).

Author: I changed it, it felt unclear for me with a link in first sentence, cause it wasn't explicitly telling about some instructions behind that link. I'd keep my version here

Backup process can be interrupted at any time. It is automatically resumed from the interruption point when restarting `vmbackup` with the same args.
Backed up data can be restored with [vmrestore](https://docs.victoriametrics.com/victoriametrics/vmrestore/).

See [this article](https://medium.com/@valyala/speeding-up-backups-for-big-time-series-databases-533c1a927883) for more details.

If you are running enterprise version, you can also use [vmbackupmanager](https://docs.victoriametrics.com/vmbackupmanager/) tool built on top of `vmbackup`. This tool simplifies
creation of hourly, daily, weekly and monthly backups.
Enterprise binaries can be downloaded and evaluated for free from [the releases page](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/latest).
See how to request a free trial license [here](https://victoriametrics.com/products/enterprise/trial/).

### Single node backup

> In order to perform a complete backup for VictoriaMetrics single node, the following command must be run:

Reviewer: ```suggestion
To make a complete backup for VictoriaMetrics single node run the following command:
```

```sh
./vmbackup -storageDataPath=</path/to/victoria-metrics-data> -snapshot.createURL=http://localhost:8428/snapshot/create -dst=gs://<bucket>/<path/to/new/backup>
```

Please refer to [full backup creation](https://docs.victoriametrics.com/victoriametrics/vmbackup/#full-backups) for further details.
We recommend using [smart backup](https://docs.victoriametrics.com/victoriametrics/vmbackup/#smart-backups) strategy as a best practice.

### Cluster backup

> In order to perform a complete backup for [VictoriaMetrics cluster](https://docs.victoriametrics.com/cluster-victoriametrics/), `vmbackup` must be run on each `vmstorage` node in cluster. Backups must

Reviewer: ```suggestion
To make a complete backup for [VictoriaMetrics cluster](https://docs.victoriametrics.com/cluster-victoriametrics/), `vmbackup` must be run on each `vmstorage` node in cluster. Backups must
```

be placed into different directories on the remote storage in order to avoid conflicts between backups from different nodes.

> For example, when creating a backup with 3 `vmstorage` nodes, the following commands must be run:

Reviewer: ```suggestion
For example, run the following command to make a backup for 3 `vmstorage` nodes:
```

* Any S3-compatible storage such as [MinIO](https://github.com/minio/minio) or [Ceph](https://docs.ceph.com/en/pacific/radosgw/s3/). See [these docs](#advanced-usage) for details.
* Local filesystem. Example: `fs://</absolute/path/to/backup>`. Note that `vmbackup` prevents from storing the backup into the directory pointed by `-storageDataPath` command-line flag, since this directory should be managed solely by VictoriaMetrics or `vmstorage`.

## Backup types

> `vmbackup` supports incremental and full backups. Incremental backups are created automatically if the destination path already contains data from the previous backup.

Reviewer 2: ```suggestion
`vmbackup` supports [incremental](https://docs.victoriametrics.com/victoriametrics/vmbackup/#incremental-backups) and [full](https://docs.victoriametrics.com/victoriametrics/vmbackup/#full-backups) backups. Incremental backups are created automatically if the destination path already contains data from the previous backup.
```

./vmbackup -storageDataPath=</path/to/victoria-metrics-data> -snapshot.createURL=http://localhost:8428/snapshot/create -dst=gs://<bucket>/<path/to/new/backup> -origin=gs://<bucket>/<path/to/existing/backup>
```

> It saves time and network bandwidth costs by performing server-side copy for the shared data from the `-origin` to `-dst`.

Author: Cause we had the same paragraph twice in the same doc - one here, another in server-side copy of the existing backup. I think it would be better that in this part we cover the fact that there's no need in upload/download and in the server-side copy we cover metadata copying vs. data copying etc

> It saves time and network bandwidth costs by performing server-side copy for the shared data from the `-origin` to `-dst`.

Reviewer 2: That's a very important statement; if overlooked, it might cause some problems. I think it's okay to at least mention it here or provide a link

The `-origin` and `-dst` must point to the same object storage bucket or to the same filesystem.

The server-side backup copy is usually performed at much faster speed comparing to the usual backup, since backup data isn't transferred
between the remote storage and locally running `vmbackup` tool. Some object storage systems might just make new names for already existing

> objects during server-side copy. However, some other are systems such as [S3 Glacier](https://aws.amazon.com/s3/storage-classes/glacier/),

Reviewer 2: maybe something like this

```
However, some systems, such as S3 Glacier, perform a full object copy during server-side copying, which can be slow and expensive. Please consult your provider's documentation for specific behavior.
```
